src/symbolize/gimli/libs_dl_iterate_phdr.rs

```diff
@@ -73,6 +86,22 @@ unsafe extern "C" fn callback(
             OsStr::from_bytes(unsafe { CStr::from_ptr(dlpi_name) }.to_bytes()).to_owned()
         }
     };
+    #[cfg(target_os = "android")]
+    let zip_offset: Option<i64> = {
+        // only check for ZIP-embedded file if we have data from /proc/self/maps
+        maps.as_ref().and_then(|maps| {
+            // check if file is embedded within a ZIP archive by searching for `!/`
+            name.as_bytes()
+                .windows(2)
+                .find(|&chunk| chunk == b"!/")
```

Comment:
Hm. We obtain this information twice. First we locate it here, then we go through `create_mapping` and then into `new_android`. I suppose it's probably less of a hassle to simply compute it twice rather than try to cart around another usize offset into something, and potentially screw THAT up.

Still, I feel like, given questions like "can files have multiple ! in them on Android?", the question arises whether this sort of "split on the `!/`" shouldn't be extracted into its own logical function?

Reply:
> probably less of a hassle to simply compute it twice rather than try to cart around another usize offset into something, and potentially screw THAT up

Yeah that was my logic.

> the question arises whether this sort of "split on the !/" shouldn't be extracted into its own logical function?

I don't have a problem with that, I'm just not sure where I'd put it. `src/symbolize/gimli.rs` maybe, or make a new sub-module?

Reply:
Yeah, putting it in gimli.rs seems fine.